libraries/chain/backing_store/db_context.cpp

```diff
@@ -0,0 +1,134 @@
+#include <eosio/chain/apply_context.hpp>
+#include <eosio/chain/exceptions.hpp>
+#include <eosio/chain/backing_store/db_context.hpp>
+
+namespace eosio { namespace chain { namespace backing_store {
+
+std::string db_context::table_event(name code, name scope, name table) {
+   return STORAGE_EVENT_ID("${code}:${scope}:${table}",
+                           ("code", code)
+                           ("scope", scope)
+                           ("table", table)
+   );
+}
+
+std::string db_context::table_event(name code, name scope, name table, name qualifier) {
+   return STORAGE_EVENT_ID("${code}:${scope}:${table}:${qualifier}",
+                           ("code", code)
+                           ("scope", scope)
+                           ("table", table)
+                           ("qualifier", qualifier)
+   );
+}
+
+void db_context::write_insert_table(fc::logger& deep_mind_logger, uint32_t action_id, name code, name scope, name table, account_name payer) {
```

Comment:
`write_` seems confusing to me. Mayb `log_` instead? `write_` makes me think this writes to the database.

Reply:
Changed